packages/gasket-plugin-https-proxy/lib/index.d.ts

```diff
@@ -0,0 +1,33 @@
+import { Plugin } from '@gasket/core';
+import type { ServerOptions as ProxyServerOptions } from 'http-proxy';
+
+interface BaseHttpsProxyConfig extends ProxyServerOptions {
+  protocol?: string;
+  /** defaults to 'localhost' */
+  hostname?: string;
+  /** defaults to 8080 */
+  port?: number;
+}
+
+declare module '@gasket/core' {
+  export type HttpsProxyConfig = RequireAtLeastOne<BaseHttpsProxyConfig, 'target' | 'forward'>;
+
+  export interface GasketConfig {
+    httpsProxy?: HttpsProxyConfig
+  }
+
+  export interface HookExecTypes {
+    httpsProxy(proxyConfig: HttpsProxyConfig): MaybeAsync<HttpsProxyConfig>,
+  }
+
+  export interface GasketActions {
+    startProxyServer: () => Promise<void>;
```

Comment:
There may be a use case for the caller to receive the proxy server object and have the action return that instead of this being a `Promise<void>`.